src/bika/lims/browser/worksheet/views/results.py

```diff
@@ -49,29 +50,28 @@ def __init__(self, context, request):
             "/++plone++senaite.core.static/assets/icons/worksheet.svg"
         )
 
-        self.layout_displaylist = WORKSHEET_LAYOUT_OPTIONS
-
     def __call__(self):
         # TODO: Refactor this function call
         showRejectionMessage(self.context)
 
         # Save the results layout
         rlayout = self.request.get("resultslayout", "")
-        if rlayout and rlayout in WORKSHEET_LAYOUT_OPTIONS.keys() \
+        if rlayout and rlayout in self.layout_displaylist.keys() \
            and rlayout != self.context.getResultsLayout():
             self.context.setResultsLayout(rlayout)
             message = _("Changes saved.")
             self.context.plone_utils.addPortalMessage(message, "info")
 
-        # Classic/Transposed View Switch
-        if self.context.getResultsLayout() == "1":
-            view = "analyses_classic_view"
-            self.Analyses = api.get_view(
-                view, context=self.context, request=self.request)
-        else:
-            view = "analyses_transposed_view"
-            self.Analyses = api.get_view(
-                view, context=self.context, request=self.request)
+        # Classic/Transposed or additional View Switch
+        view = self.context.getResultsLayout()
+        if view not in self.layout_displaylist.keys():
+            message = _("Layout view '{}' not found. Set '{}' as layout view.".format(view, DEFAULT_WORKSHEET_LAYOUT))
```

Comment:
https://github.com/senaite/senaite.core/pull/1977#discussion_r857580612

Maybe this block of code resolve problem?

This condition was added intentionally for the "first" check and subsequent possibly erroneous